From b036279e98a584e5f55c98963b904b76b7f5e0b8 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@freescale.com>
Date: Thu, 29 Jan 2015 10:32:32 +0800
Subject: [PATCH 0255/5242] MLK-10174-2 usb: chipidea: host: keep controller
 power if host has dev connected

commit  19acd57625be8f832a46ca113a4517c79d8af718 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch is to keep usb controller power on while system suspend if there
is usb device connected by telling GPC to keep mega fast domain power.

Signed-off-by: Li Jun <jun.li@freescale.com>
(cherry picked from commit 74d1524208bfa052f9adac28c8f7d8fb8ed78499)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/chipidea/host.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/usb/chipidea/host.c b/drivers/usb/chipidea/host.c
index cdce7e8..0ed3aba 100644
--- a/drivers/usb/chipidea/host.c
+++ b/drivers/usb/chipidea/host.c
@@ -13,6 +13,7 @@
 #include <linux/usb/hcd.h>
 #include <linux/usb/chipidea.h>
 #include <linux/regulator/consumer.h>
+#include <linux/imx_gpc.h>
 
 #include "../host/ehci.h"
 
@@ -461,11 +462,16 @@ static void ci_hdrc_host_restore_from_power_lost(struct ci_hdrc *ci)
 
 static void ci_hdrc_host_suspend(struct ci_hdrc *ci)
 {
+	if (ci_hdrc_host_has_device(ci))
+		imx_gpc_mf_request_on(ci->irq, 1);
+
 	ci_hdrc_host_save_for_power_lost(ci);
 }
 
 static void ci_hdrc_host_resume(struct ci_hdrc *ci, bool power_lost)
 {
+	imx_gpc_mf_request_on(ci->irq, 0);
+
 	if (power_lost)
 		ci_hdrc_host_restore_from_power_lost(ci);
 }
-- 
1.7.9.5

